src/vanilla/Model/MethodTS.cs

```diff
@@ -45,21 +45,40 @@ public enum MethodFlavor
             HttpOperationResponse
         }
 
+        private const string baseHttpOperationResponseName = "msRest.BaseHttpOperationResponse";
+
         public string HttpOperationResponseName
         {
             get
             {
-                if (ReturnType.Headers == null)
+                if (ReturnType.Headers != null)
+                {
+                    return Regex.Replace(ReturnType.Headers.Name, "Headers$", "Response");
+                }
+                else if (ReturnType.Body != null)
                 {
-                    return $"msRest.HttpOperationResponse<{ReturnTypeTSString}>";
+                    return MethodGroup.Name.Value.ToPascalCase() + Name.Value.ToPascalCase() + "Response";
                 }
                 else
                 {
-                    return $"Models.{Regex.Replace(ReturnType.Headers.Name, "Headers$", "Response")}";
+                    return baseHttpOperationResponseName;
                 }
             }
         }
 
+        public string HttpOperationResponseReferenceName
+        {
+            get
+            {
+                string responseName = HttpOperationResponseName;
+                return responseName == baseHttpOperationResponseName
+                    ? baseHttpOperationResponseName
+                    : "Models." + responseName;
+            }
+        }
+
+        public bool HasCustomHttpResponseType => HttpOperationResponseName != baseHttpOperationResponseName;
```

Comment:
This property's value is very close to the logic on [line 551](https://github.com/Azure/autorest.typescript/pull/248/files#diff-a0945a4f62f5a6e958b85ad35db7086eR551,) but slightly different. Can you somehow point out the difference between these two expressions (either with a new property or a comment)?

Reply:
I can just fix line 551 to use this property. The difference in practice is nil.

Reply:
That's not true, though, right? I thought that the property didn't include the `msRest.` prefix?

Reply:
The base response is always referenced as `msRest.BaseHttpOperationResponse`. The custom responses are what get prefixed at the usage site.